File: src/main/java/mods/betterfoliage/client/KeyHandler.java
package mods.betterfoliage.client;

import mods.betterfoliage.BetterFoliage;
import mods.betterfoliage.client.gui.ConfigGuiFactory;
import net.minecraft.client.settings.KeyBinding;
import net.minecraftforge.fml.client.FMLClientHandler;
import net.minecraftforge.fml.client.registry.ClientRegistry;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.InputEvent;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class KeyHandler {
	
	public static KeyBinding guiBinding;
	
	public KeyHandler() {
		guiBinding = new KeyBinding("key.betterfoliage.gui", 66, BetterFoliage.MOD_NAME);
		ClientRegistry.registerKeyBinding(guiBinding);
	}
	
	@SubscribeEvent
	public void handleKeyPress(InputEvent.KeyInputEvent event) {
		if (guiBinding.isPressed()) FMLClientHandler.instance().showGuiScreen(new ConfigGuiFactory.ConfigGuiBetterFoliage(null));
	}
}

File: src/main/kotlin/mods/octarinecore/client/KeyHandler.kt
package mods.octarinecore.client

import net.minecraft.client.settings.KeyBinding
import net.minecraftforge.client.event.InputEvent
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.client.registry.ClientRegistry

class KeyHandler(val modId: String, val defaultKey: Int, val lang: String, val action: (InputEvent.KeyInputEvent)->Unit) {

    val keyBinding = KeyBinding(lang, defaultKey, modId)

    init {
        ClientRegistry.registerKeyBinding(keyBinding)
        MinecraftForge.EVENT_BUS.register(this)
    }

    @SubscribeEvent
    fun handleKeyPress(event: InputEvent.KeyInputEvent) {
        if (keyBinding.isPressed) action(event)
    }
}